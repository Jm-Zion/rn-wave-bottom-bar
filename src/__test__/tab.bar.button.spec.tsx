import React from 'react';
import { View } from 'react-native';
import * as renderer from 'react-test-renderer';

import { BarButton, TabBarButton } from '../components/tab.bar.button';

const tabBarIcon = (props: any) => <View {...props} />;

describe('TabBarButton', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  afterAll(() => {
    jest.useRealTimers();
  });
  it("Should render well when it's Focus", () => {
    const tree = renderer.create(
      <BarButton
        mode="default"
        index={0}
        isFocused
        options={{ tabBarIcon }}
        onPress={jest.fn()}
        onLongPress={jest.fn()}
      />
    );
    jest.runAllTimers();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('Should render well it is not focused', () => {
    const tree = renderer.create(
      <BarButton
        mode="default"
        index={0}
        isFocused={false}
        options={{ tabBarIcon }}
        onPress={jest.fn()}
        onLongPress={jest.fn()}
      />
    );
    jest.runAllTimers();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it("Should render well when it's Focus", () => {
    const tree = renderer.create(
      <TabBarButton
        mode="default"
        index={0}
        isFocused
        options={{ tabBarIcon }}
        onPress={jest.fn()}
        onLongPress={jest.fn()}
      />
    );
    jest.runAllTimers();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('Should render well it is not focused', () => {
    const tree = renderer.create(
      <TabBarButton
        mode="default"
        index={0}
        isFocused={false}
        options={{ tabBarIcon }}
        onPress={jest.fn()}
        onLongPress={jest.fn()}
      />
    );
    jest.runAllTimers();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('Should render well it is not focused', () => {
    const tree = renderer.create(
      <TabBarButton
        mode="square"
        index={0}
        isFocused={false}
        options={{ tabBarIcon }}
        onPress={jest.fn()}
        onLongPress={jest.fn()}
      />
    );
    jest.runAllTimers();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
