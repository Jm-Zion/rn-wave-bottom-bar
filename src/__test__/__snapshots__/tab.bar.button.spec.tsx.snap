// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarButton Should render well it is not focused 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "elevation": 12,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": "100%",
        "zIndex": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "zIndex": 12,
        }
      }
    >
      <View
        color="white"
        focused={false}
        size={28}
      />
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well it is not focused 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 32,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      color="white"
      focused={false}
      size={28}
    />
  </View>
</View>
`;

exports[`TabBarButton Should render well it is not focused 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 16,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      color="white"
      focused={false}
      size={28}
    />
  </View>
</View>
`;

exports[`TabBarButton Should render well when it's Focus 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "elevation": 12,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
        "width": "100%",
        "zIndex": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "zIndex": 12,
        }
      }
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`TabBarButton Should render well when it's Focus 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 12,
      "flex": 1,
      "justifyContent": "center",
      "zIndex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 32,
        "elevation": 10,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": -18,
          },
        ],
        "width": 60,
        "zIndex": -1,
      }
    }
  >
    <View
      color="white"
      focused={true}
      size={28}
    />
  </View>
</View>
`;
